snuba/snuba_migrations/generic_metrics/0031_counters_meta_table_mv.py

```diff
@@ -0,0 +1,71 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import AggregateFunction, Column, DateTime, String, UInt
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+from snuba.migrations.operations import OperationTarget
+from snuba.utils.schemas import Float
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+    view_name = "generic_metric_counters_meta_aggregation_mv"
+    dest_table_name = "generic_metric_counters_meta_aggregated_local"
+    dest_table_columns: Sequence[Column[Modifiers]] = [
+        Column("org_id", UInt(64)),
+        Column("project_id", UInt(64)),
+        Column("use_case_id", String(Modifiers(low_cardinality=True))),
+        Column("metric_id", UInt(64)),
+        Column("tag_key", String()),
+        Column("tag_value", String()),
+        Column("timestamp", DateTime(modifiers=Modifiers(codecs=["DoubleDelta"]))),
+        Column("retention_days", UInt(16)),
+        Column("value", AggregateFunction("sum", [Float(64)])),
+    ]
+    storage_set_key = StorageSetKey.GENERIC_METRICS_COUNTERS
+
+    def forwards_ops(self) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.CreateMaterializedView(
+                storage_set=self.storage_set_key,
+                view_name=self.view_name,
+                columns=self.dest_table_columns,
+                destination_table_name=self.dest_table_name,
+                target=OperationTarget.LOCAL,
+                query="""
+                SELECT
+                    org_id,
+                    project_id,
+                    use_case_id,
+                    metric_id,
+                    tag_key,
+                    tag_value,
+                    toStartOfWeek(timestamp) as timestamp,
+                    retention_days,
+                    sumState(count_value) as count
+                FROM generic_metric_counters_raw_local
+                ARRAY JOIN
+                    tags.key AS tag_key,
+                    tags.raw_value AS tag_value
+                GROUP BY
+                    org_id,
+                    project_id,
+                    use_case_id,
+                    metric_id,
+                    tag_key,
+                    tag_value,
+                    timestamp,
+                    retention_days
```

Comment:
could we make this `max(retention_days, 90)` or whatever the correct syntax is? with the upcoming 180-day billing use case I don't want stuff sticking around that long here.

Reply:
Why not? Wouldn't they also want (in theory) to autocomplete on their metrics and tag keys? This is meant so that meta data stays as long as the actual data stays (roughly).

Reply:
As I think about this, I'm not sure we should support autocomplete for all use cases by default. For example, storing all this data for escalating issues is a waste. Tag values are going to be issue group IDs and there's never any reason to autocomplete them.

I don't want to block you from validating this overall approach, but I think we should consider adding a boolean column to the raw table like `record_meta` and only materialize rows where record_meta = true. It's just wasted space for use cases that never have manual user interaction.

Reply:
I agree that we should make recording information in the meta table configurable. Especially since there would be more things coming on the generic metrics pipeline like billing information where auto complete is not a requirement. We can reduce the cost of operating the generic metrics pipeline by restricting the feature to only the use cases which need them.

Reply:
How do I make this configurable? I can add an exclusion to the materialized table query (`WHERE use_case_id NOT IN (...)` but is there a way to make that configurable?